File: 238. Product of Array Except Self/main.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        n= len(nums)
        arr=[1]*n

        mul=1
        for i in range(n):
            arr[i]=mul
            mul=mul*nums[i]

        mul=1
        for i in range(n-1,-1,-1):
            arr[i]=arr[i]*mul
            mul=mul*nums[i]

        
        return arr

        

File: 238. Product of Array Except Self/main.cpp
#include <bits/stdc++.h>

class Solution {
    

public:
    std::vector<int> productExceptSelf(std::vector<int>& nums) {
        int n=nums.size();

        std::vector<int> arr(n,1);


        int mul=1;
        for(int i=0;i<n;i++)
        {
            arr[i]=mul;
            mul*=nums[i];


        }
        mul=1;
        for(int i=n-1;i>=0;i--)
        {
            arr[i]=mul*arr[i];
            mul*=nums[i];
        }

         return arr;
        

    }
};